evaluation_scripts/plot_functions.py

```diff
@@ -0,0 +1,296 @@
+# This script contains the functions used for plotting different plots in
+# other scripts used for evaluation.
+
+# Author: Shweta Narkhede and Camilo Salcedo
+# Created on: Oct 24th, 2020
+# %%
+import pandas as pd
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+import eval_functions as ef
+import seaborn as sns
+
+#%% Functions
+
+def get_histogram(forecasts1, obs_week, title_string):
+    """Get Histograms:
+    -----------------------------------
+    This function plots histograms of predicted weekly flow data and
+    the count of the flow value prediction. It also plots observed weekly
+    data for last week for comaparision.
+    -----------------------------------
+    Parameters:
+    forecasts1  = array
+                every student's forecast for week 1
+    obs_week    = float
+                provides week's observed flow
+    title_string = string
+                   provides the title of the histogram
+    -----------------------------------
+    Outputs:
+    figure of Histogram plot
+    """
+    plt.figure(figsize=(8, 6))
+    plt.hist(forecasts1, bins=120, color='blue', alpha=0.75,
+             label='Student Guesses')
+    histogram = plt.plot([obs_week]*3, np.arange(0, 3, 1), color='red',
+                         linestyle='-', label='Actual mean')
+    plt.title(title_string)
+    plt.xlabel('Flow Forecast (cfs)')
+    plt.ylabel('Count')
+    plt.legend(loc='upper left')
+    return histogram
+
+
+def get_simpleplot(forecasts, class_avg, obs_week, title_string):
+    """Get Simple plot:
+    ------------------------------------
+    This function plots a simple line plot of student's weekly averaged
+    forecast for a week
+    ------------------------------------
+    Parameters:
+    forecasts = array
+                provides weekly forecasted flow of each student
+    class_avg = float
```

Comment:
Could you just calculate the average within your function rather than requiring the user to do this outside the function?

Reply:
Thanks for your suggestion. We modified the function, and now the average is calculated within the function and plotted.